cmd/plume/prerelease.go

```diff
@@ -86,17 +105,70 @@ func init() {
 	sort.Sort(sort.StringSlice(platformList))
 
 	cmdPreRelease.Flags().StringSliceVar(&selectedPlatforms, "platform", platformList, "platform to pre-release")
+	cmdPreRelease.Flags().StringVar(&selectedSystem, "system", "cl", "system to pre-release")
 	cmdPreRelease.Flags().StringVar(&azureProfile, "azure-profile", "", "Azure Profile json file")
 	cmdPreRelease.Flags().StringVar(&awsCredentialsFile, "aws-credentials", "", "AWS credentials file")
 	cmdPreRelease.Flags().StringVar(&verifyKeyFile,
 		"verify-key", "", "path to ASCII-armored PGP public key to be used in verifying download signatures.  Defaults to CoreOS Buildbot (0412 7D0B FABE C887 1FFB  2CCE 50E0 8855 93D2 DCB4)")
 	cmdPreRelease.Flags().StringVar(&imageInfoFile, "write-image-list", "", "optional output file describing uploaded images")
 
 	AddSpecFlags(cmdPreRelease.Flags())
+	AddFedoraSpecFlags(cmdPreRelease.Flags())
 	root.AddCommand(cmdPreRelease)
 }
 
-func runPreRelease(cmd *cobra.Command, args []string) error {
+func runFedoraPreRelease(system string, cmd *cobra.Command, args []string) error {
+	if len(args) > 0 {
+		return errors.New("no args accepted")
+	}
+
+	for _, platformName := range selectedPlatforms {
+		if _, ok := platforms[platformName]; !ok {
+			return fmt.Errorf("Unknown platform %q", platformName)
+		}
+	}
+
+	spec, err := ChannelFedoraSpec()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	client := http.Client{}
+
+	var imageInfo imageInfo
+
+	imageMetadata := map[string]interface{}{
+		"Env":       specEnv,
+		"Version":   specFedoraVersion,
+		"Timestamp": specTimestamp,
+		"Respin":    specRespin,
+		"ImageType": specImageType,
+		"Arch":      specArch,
+	}
+
+	for _, platformName := range platformList {
```

Comment:
I know this is most likely copy-pasted from the existing code but figured I'd post this up anyways.

Given we've already validated the `selectedPlatforms` list it seems like this could be simplified to:
```golang
for _, platformName := range selectedPlatforms {
    platform := platforms[platformName]
    plog.Printf("Running %v pre-release...", platform.displayName)
    if err := platform.handler(ctx, system, &client, nil, &spec, &imageInfo, &imageMetadata); err != nil {
        return nil
    }
}
```

Reply:
This is now running against all platforms in `platformList` rather than the list of selected platforms. Same with the repeat block in `runCLPreRelease`